src/gen.cpp

```diff
@@ -520,15 +522,26 @@ int generate_cmake(const char *path, bool root) {
         }
     }
 
-    if (!cmake.vcpkg.version.empty()) {
-        assert("pmm is required in fetch-content for vcpkg to work" && cmake.contents.count("pmm") != 0);
-        comment("Bootstrap vcpkg");
-        cmd("include")("${pmm_SOURCE_DIR}/pmm.cmake");
-        tsl::ordered_map<std::string, std::vector<std::string>> vcpkg_args;
-        vcpkg_args["REVISION"] = {cmake.vcpkg.version};
-        vcpkg_args["REQUIRES"] = cmake.vcpkg.packages;
-        auto vcpkg = std::make_pair("VCPKG", vcpkg_args);
-        cmd("pmm")(vcpkg).endl();
+    if (!cmake.vcpkg.packages.empty()) {
+        cmd("include")("FetchContent");
+        cmd("message")("STATUS", "Fetching vcpkg...");
+        cmd("FetchContent_Declare")("vcpkg", "URL", "https://github.com/microsoft/vcpkg/archive/refs/tags/2021.05.12.tar.gz");
+        cmd("FetchContent_MakeAvailable")("vcpkg");
+        cmd("include")("${vcpkg_SOURCE_DIR}/scripts/buildsystems/vcpkg.cmake");
+        using namespace nlohmann;
+        json j;
+        j["$schema"] = "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json";
+        j["name"] = cmake.project_name;
+        if (cmake.project_version.empty())
+            throw std::runtime_error("vcpkg manifest mode requires that the project have a version string!");
```

Comment:
I don't think this is true, for me it was working fine without the version field

Reply:
Hmm I was going with their docs, I've removed the requirement.

Reply:
Yeah the docs are right, just tried it and it fails. Currently in the process adding some tests and I'll fix it

Reply:
I found that `"version-string": ""` is working so I'll just unconditionally set that